catalogs/catalog-hive/src/main/java/com/datastrato/gravitino/catalog/hive/HiveTableOperations.java

```diff
@@ -99,7 +107,91 @@ private String[][] getFieldNames(String partitionName) {
 
   @Override
   public Partition addPartition(Partition partition) throws PartitionAlreadyExistsException {
-    throw new UnsupportedOperationException();
+    Preconditions.checkArgument(
+        partition instanceof IdentityPartition, "Hive only supports identity partition");
+    IdentityPartition identityPartition = (IdentityPartition) partition;
+
+    List<String> transformFields =
+        Arrays.stream(table.partitioning())
+            .map(t -> ((Transforms.IdentityTransform) t).fieldName()[0])
```

Comment:
Does here mean that hive doesn't support nested column for partitioning, right?

Reply:
Also is it better to use `Set`, since you need to check the "contains" below?

Reply:
> Does here mean that hive doesn't support nested column for partitioning, right?

yep, see [source code](https://github.com/apache/hive/blob/1dc3dc9ede4fa932df6d5a5626f000e720481188/common/src/java/org/apache/hadoop/hive/ql/ErrorMsg.java#L203)

Reply:
> Also is it better to use `Set`, since you need to check the "contains" below?

fixed